### PR TITLE
add transaction component

### PR DESCRIPTION
**Pull request by lilyanB**
image

similare to this component called "Selector":

image

### CODE REVIEW

File: src/components/Transaction/Transaction.test.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+        '0x71c081f64f7bfe8ecb486029875f8e088db220649c62ab2c253da6c471ae137b',
+      posIcon: <FiAlertTriangle />,
+      amount: '0,00',
+      timestamp: '2021-08-20T15:00:00.000Z',
+    };
+    render(<Transaction {...args} />);
+
+    let transaction = screen.getByTestId('transaction');
+
+    expect(transaction).toBeTruthy();
```
Comment (mazmassa) [inline src/components/Transaction/Transaction.test.tsx:20]:
```suggestion
    expect(transaction).toBeInTheDocument();
```

File: src/components/Transaction/Transaction.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+export interface TransactionProps extends ComponentPropsWithoutRef<'div'> {
+  preIcon: JSX.Element;
+  title: string;
+  address: string;
+  posIcon: JSX.Element;
+  amount: string;
+  timestamp: string;
+  customClass?: string;
+  variant?: 'secondary' | undefined;
```
Comment (mazmassa) [inline src/components/Transaction/Transaction.tsx:15]:
```suggestion
  variant?: 'secondary' | 'primary' | undefined;
```

File: src/components/Transaction/Transaction.stories.tsx [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+  preIcon: <FiArrowUpRight />,
+  title: 'Transaction category',
+  address: '0x71c081f64f7bfe8ecb486029875f8e088db220649c62ab2c253da6c471ae137b',
+  posIcon: <MassaLogo size={24} />,
+  amount: '0,00',
+  timestamp: '2021-08-20T15:00:00.000Z',
+};
+
+export const _Primary = {
+  render: () => <Transaction {...args} customClass="!w-1/2" />,
```
Comment (mazmassa) [inline src/components/Transaction/Transaction.stories.tsx:19]: Is there any special reason force the override with important here!? **!w-1/2** 🤔

File: src/components/Transaction/Transaction.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  const date = new Date(timestamp);
+  const options: Intl.DateTimeFormatOptions = {
+    weekday: 'long',
+    hour: '2-digit',
+    minute: '2-digit',
+    hour12: true,
+  };
+  const formattedDate = date
+    .toLocaleString('en-US', options)
+    .replace(/(\w+)(\s+)/, '$1, ');
```
Comment (mazmassa) [inline src/components/Transaction/Transaction.tsx:66]: I think we can deal with date here as simple string and let's the component where we place the `<Trasaction>` deal with the format of it.

File: src/components/Transaction/Transaction.stories.tsx (hunk 3 above)
Comment (lilyanB) [reply src/components/Transaction/Transaction.stories.tsx:19]: It's just like "selector" component

File: src/components/Transaction/Transaction.test.tsx (hunk 1 above)
Comment (lilyanB) [reply src/components/Transaction/Transaction.test.tsx:20]: [done](url)

File: src/components/Transaction/Transaction.tsx (hunk 4 above)
Comment (lilyanB) [reply src/components/Transaction/Transaction.tsx]: done

File: src/components/Transaction/Transaction.test.tsx (hunk 1 above)
Comment (lilyanB) [reply src/components/Transaction/Transaction.test.tsx:20]: I have this error with "toBeInTheDocument" :

```
● Components | Buttons | Transaction › it should render

    TypeError: expect(...).toBeInTheDocument is not a function

      18 |     let transaction = screen.getByTestId('transaction');
      19 |
    > 20 |     expect(transaction).toBeInTheDocument();
         |                         ^
      21 |   });
      22 | });
      23 |
```

Comment (maxime-carabina) [reply src/components/Transaction/Transaction.test.tsx:20]: > I have this error with "toBeInTheDocument" :
> 
> ```
> ● Components | Buttons | Transaction › it should render
> 
>     TypeError: expect(...).toBeInTheDocument is not a function
> 
>       18 |     let transaction = screen.getByTestId('transaction');
>       19 |
>     > 20 |     expect(transaction).toBeInTheDocument();
>          |                         ^
>       21 |   });
>       22 | });
>       23 |
> ```

Add this at the top of the file :

`import '@testing-library/jest-dom';`

**Review by maxime-carabina — APPROVED** [review]

File: src/components/Transaction/Transaction.test.tsx (hunk 1 above)
Comment (lilyanB) [reply src/components/Transaction/Transaction.test.tsx:20]: done

File: src/components/Transaction/Transaction.stories.tsx (hunk 3 above)
Comment (mazmassa) [reply src/components/Transaction/Transaction.stories.tsx:19]: Oh I see! I think you can remove than. It will take the whole place on `width`

Comment (lilyanB) [reply src/components/Transaction/Transaction.stories.tsx]: okai, done

**Pull request merged**